src/App.tsx

```diff
@@ -16,80 +17,171 @@ export const goodsFromServer = [
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+  reset: boolean,
+  alphabet: boolean,
+  length: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+    reset: false,
+    alphabet: false,
+    length: false,
+  };
+
+  reverse = () => {
+    this.setState(state => {
+      return {
+        isReversed: !state.isReversed,
+      };
+    });
+  };
+
+  sortAlphabet = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+      reset: true,
+      alphabet: true,
+      length: false,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+      reset: true,
+      alphabet: false,
+      length: true,
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+      reset: false,
+      alphabet: false,
+      length: false,
+    });
+  };
+
+  render() {
+    const {
+      isReversed,
+      sortType,
+      reset,
+      alphabet,
+      length,
+    } = this.state;
+    const data = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              classNames('button is-info', { 'is-light': !alphabet })
+            }
+            onClick={() => {
+              this.sortAlphabet();
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames('button is-success', { 'is-light': !length })
+            }
+            onClick={() => {
+              this.sortByLength();
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames('button is-warning', { 'is-light': !isReversed })
+            }
+            onClick={() => {
+              this.reverse();
+            }}
+          >
+            Reverse
+          </button>
+
+          {
+            (reset || isReversed)
```

Comment:
```suggestion
            (sortType !== SortType.NONE || isReversed)
```

here we can also simply check the state of the sort type